src/arkhamdb/DeckImporterMain.ttslua

```diff
@@ -53,6 +53,30 @@ function buildDeck(playerColor, deckId)
       loadCards)
 end
 
+-- Modelled after onDeckResult from arkhamdb/ArkhamDb.ttslua, but with some necessary changes
+function spawnDeckFromToken(params)
+  local deckData = JSON.decode(params.coin.getGMNotes())
+    destroyObject(params.coin)
+    deckData["playerColor"] = params.playerColor
+    -- Add Investigator Cards
+    deckData["slots"][deckData.investigatorId] = 1
+    deckData["slots"][string.sub(deckData.investigatorId, 1, 5) .. "-m"] = 1
+    -- Add Bonded Cards
+    for bondedId, bondedCount in pairs(deckData.bondedList) do
+      deckData.slots[bondedId] = bondedCount
+    end
+    -- Add Customizable Upgrade Sheets
+    
+    -- Add Summon Servitor minicard
+    if deckData.slots["09080"] ~= nil then
+      deckData.slots["09080-m"] = 1
+    end
+    -- Add On The Mend
+
+
+    loadCardsHelper(deckData)
+end
```

Comment:
I think it would be more future-proof to combine this with the existing spawn deck function instead of repeating the same process

Reply:
The issue with this is that it would require formatting the deck data into the horrible way that it is formatted when receiving it from ArkhamDB, which is something I would rather avoid 😅

Reply:
It could also mean to format the horrible data into this format ;)
We should just avoid having two "data handling" methods that need updating (like the addition of "On the Mend", "Summoned Servitor" etc.)